### ISSUE TITLE
Add converters/*jections for between scala types and Tensor

### ISSUE THREAD

**Issue by ravwojdyla** [spotify/scio#679]
Related to TensorFlow API - would be nice to provide converters/(bi/in)jections for common types (arrays) to Tensor to make it easier to build Tensor objects.

**Comment by richwhitjr** [comment]
Should this be in Scio or maybe instead in Bijection or similar library?  Seems useful outside of Scio also.

**Comment by nevillelyh** [comment]
It's easy to support primitive arrays (bool, int, long, float, double) but do we wanna support strings too? Or treat them as `Array[Byte]`?

Probably easy to add some implicit methods so we can do:
`intArray.asTensor` with optional shap, and `tensor.asInts`?

**Comment by nevillelyh** [comment]
Leaning towards closing this since it feels out of scope. Plus whatever API we come up with might make it more confusing for users?

**Comment by ravwojdyla** [comment]
Closing for now, pending on redesign of `predict` method.
